cmd/snap/cmd_run.go

```diff
@@ -448,13 +448,25 @@ func (x *cmdRun) snapRunTimer(snapApp, timer string, args []string) error {
 
 var osReadlink = os.Readlink
 
-func isReexeced() bool {
+// snapdHelperPath return the path of a helper like "snap-confine" or
+// "snap-exec" based on if snapd is re-execed or not
+func snapdHelperPath(toolName string) (string, error) {
```

Comment:
is this helper just a refactor or there is something significative happening here that am I missing?

Reply:
Its a refactor mostly, we only need the isReexeced() to find what tool to run so extracting this all into a common function seems wise. Plus in another place the code is actually hardcoding "core/current" which of course is wrong.